cedar-policy-core/src/entities/json/err.rs

```diff
@@ -14,72 +14,97 @@
  * limitations under the License.
  */
 
+use std::fmt::Display;
+
 use super::SchemaType;
 use crate::ast::{
     EntityType, EntityUID, Expr, ExprKind, Name, RestrictedExpr, RestrictedExpressionError,
 };
 use crate::extensions::ExtensionsError;
+use crate::parser::err::ParseErrors;
 use smol_str::SmolStr;
 use thiserror::Error;
 
+/// Escape kind
+#[derive(Debug)]
+pub enum EscapeKind {
+    /// Escape `__expr`
+    Expr,
+    /// Escape `__entity`
+    Entity,
+    /// Escape `__extn`
+    Extension,
+}
+
+impl Display for EscapeKind {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Self::Entity => write!(f, "__entity"),
+            Self::Expr => write!(f, "__expr"),
+            Self::Extension => write!(f, "__extn"),
+        }
+    }
+}
+
 /// Errors thrown during deserialization from JSON
 #[derive(Debug, Error)]
 pub enum JsonDeserializationError {
-    /// Error thrown by `serde_json`
+    /// Error thrown by the `serde_json` crate
     #[error("{0}")]
     Serde(#[from] serde_json::Error),
-    /// Contents of an `__expr` escape failed to parse as a Cedar expression.
+    /// Contents of an escape failed to parse.
     ///
-    /// `__expr` is deprecated (starting with the 1.2 release), and once it is
-    /// removed, this error will also be removed.
-    #[error(transparent)]
-    ExprParseError(crate::parser::err::WithContext),
-    /// Contents of an `__entity` escape failed to parse as an entity reference
-    #[error(transparent)]
-    EntityParseError(crate::parser::err::WithContext),
-    /// Function name in an `__extn` escape failed to parse as an extension function name
-    #[error(transparent)]
-    ExtnParseError(crate::parser::err::WithContext),
+    /// Note that `__expr` is deprecated and once it is
+    /// removed, this error will also be removed
+    #[error("failed to parse escape `{kind}`: {value}, errors: {errs}")]
+    ParseEscape {
+        /// Escape kind
+        kind: EscapeKind,
+        /// Escape value at fault
+        value: String,
+        /// Parse errors
+        errs: ParseErrors,
+    },
     /// Restricted expression error
     #[error(transparent)]
     RestrictedExpressionError(#[from] RestrictedExpressionError),
     /// Error thrown by an operation on `Extensions`
     #[error(transparent)]
     ExtensionsError(#[from] ExtensionsError),
     /// A field that needs to be a literal entity reference, was some other JSON value
-    #[error("{ctx}, expected a literal entity reference, but got {got}")]
+    #[error("{ctx}, expected a literal entity reference, but got `{got}`")]
```

Comment:
This would be more similar to our other error messages:

```suggestion
    #[error("{ctx}, expected a literal entity reference, but got: {got}")]
```